pkg/controllers/vdb/createarchive_reconciler.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/createarchive"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/saverestorepoint"
+	"github.com/vertica/vertica-kubernetes/pkg/vdbstatus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateArchiveReconciler will create archive and save restore point if triggered
+type CreateArchiveReconciler struct {
+	VRec *VerticaDBReconciler
+	Vdb  *vapi.VerticaDB
+	Log  logr.Logger
+	client.Client
+	Dispatcher     vadmin.Dispatcher
+	PFacts         *PodFacts
+	OriginalPFacts *PodFacts
+	InitiatorIP    string // The IP of the pod that we run vclusterOps from
+}
+
+func MakeCreateArchiveReconciler(r *VerticaDBReconciler, vdb *vapi.VerticaDB, log logr.Logger,
+	pfacts *PodFacts, dispatcher vadmin.Dispatcher, cli client.Client) controllers.ReconcileActor {
+	pfactsForMainCluster := pfacts.Copy(vapi.MainCluster)
+	return &CreateArchiveReconciler{
+		VRec:           r,
+		Log:            log.WithName("CreateArchiveReconciler"),
+		Vdb:            vdb,
+		Client:         cli,
+		Dispatcher:     dispatcher,
+		PFacts:         &pfactsForMainCluster,
+		OriginalPFacts: pfacts,
+	}
+}
+
+// Reconcile will create an archive if the status condition indicates true
+// And will save restore point to the created arcihve if restorePoint.archive value
+// is provided in the CRD spec
+func (c *CreateArchiveReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	err := c.PFacts.Collect(ctx, c.Vdb)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// No-op if no database exists
+	if !c.PFacts.doesDBExist() {
+		return ctrl.Result{}, nil
+	}
+
+	hostIP, ok := c.PFacts.FindFirstUpPodIP(true, "")
+	if !ok {
+		// If no running pod found, then there is nothing to do and we can just continue on
+		return ctrl.Result{}, nil
+	}
+
+	// Only proceed if the SaveRestorePointsNeeded status condition is set to true.
+	if c.Vdb.IsStatusConditionTrue(vapi.SaveRestorePointsNeeded) {
+		if c.Vdb.Spec.RestorePoint != nil && c.Vdb.Spec.RestorePoint.Archive != "" {
+			// Always tried to create archive
+			// params: context, host, archive-name, sandbox, num of restore point(0 is unlimited)
+			err = c.runCreateArchiveVclusterAPI(ctx, hostIP, c.Vdb.Spec.RestorePoint.Archive, "", 0)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			// Once save restore point, change condition
+			// params: context, host, archive-name, sandbox, num of restore point(0 is unlimited)
+			err = c.runSaveRestorePointVclusterAPI(ctx, hostIP, c.Vdb.Spec.RestorePoint.Archive, "")
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		// archinve name param not set correctly, Log warning
+		c.VRec.Eventf(c.Vdb, corev1.EventTypeWarning, events.CreateArchiveFailed,
```

Comment:
A simple log message will be enough.

Reply:
Switched to log msg in https://github.com/vertica/vertica-kubernetes/pull/918/commits/08a8a4944bb784afbe64019e2ee78c2c246b5f90